backend/backend_factory.py

```diff
@@ -0,0 +1,10 @@
+import backend
+
+class BackendFactory:
+    
+    @staticmethod
+    def create(width: int, height: int, publish_recording_status_func) -> backend.Backend:
```

Comment:
I don't like having to pass around all this constructor stuff, from the factory, to the derived class, then to the `super()` base class... @khanguslee , is there a cool pattern I should be using? I was wondering about a builder but I'm really not sure. Maybe I'm just not pythoning correctly either.

Reply:
A quick google shows that it is acceptable to pass in params like this in this case. As long as we're not passing too many params!

ALSO, for this to be a proper factory class, you need to pass in the factory type as a parameter rather than importing that value from elsewhere. (So `Overlay` `connect()` should work out if it is on a pi or not a pi.